### PR TITLE
Drive AppDesigner folder from project properties

### PR DESCRIPTION
**Pull request by davkean**
- Removed PropertiesFolderProjectTreePropertiesProvider and MyProjectFolderProjectTreePropertiesProvider and merged into a single provider
    - Behavior differences are driven by `<AppDesignerFolder>` and `<AppDesignerFolderContentsVisibleOnlyInShowAllFiles>` properties

- Implement IProjectTreeSettingsProvider on AppDesignerFolderProjectTreePropertiesProvider, which makes the PhysicalTreeProvider aware that it needs to call us back every time the AppDesignerFolder/AppDesignerFolderContentsVisibleOnlyInShowAllFiles properties changes
    - This enables us to now add a rename tracker that when the properties folder changes, we fix update the name of the property, and the tree automatically sees the new value

### CODE REVIEW

**Comment by davkean** [comment]
tag @dotnet/roslyn-analysis

**Comment by davkean** [comment]
This fixes: https://github.com/dotnet/roslyn-project-system/issues/46

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AppDesignerFolderProjectTreePropertiesProvider.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Imaging;
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AppDesignerFolderProjectTreePropertiesProvider.cs]: Sort usings.

**Comment by srivatsn** [comment]
👍

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AppDesignerFolderProjectTreePropertiesProvider.cs (hunk 1 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/AppDesignerFolderProjectTreePropertiesProvider.cs]: :+1:

**Pull request merged**